include/boost/url/impl/url_base.hpp

```diff
@@ -22,194 +22,48 @@ segments_type(
 {
 }
 
-url_view::
-params_type::
-params_type(
-    url_base const& v) noexcept
-    : s_(v.s_)
-    , pt_(&v.pt_)
+bool
+url_base::
+segments_type::
+iterator::
+operator==(
+    iterator other) const noexcept
 {
+    BOOST_ASSERT(
+        v_ != other.v_ ||
+        off_ != other.off_ ||
+        n_ == other.n_);
+    return
+        v_ == other.v_ &&
```

Comment:
C++ standard leaves comparison of iterators from different containers as undefined. Thus this code is not quite correct. We should assert that the containers are the same. And when we return the value we should not check that the containers are the same.